eng/install-native-dependencies.sh

```diff
@@ -1,5 +1,10 @@
 #!/usr/bin/env bash
 
+if [ "$1" = "OSX" ] && [ "$2" = "arm64" ]; then
```

Comment:
Not sure this is a great place to begin doing this. This should probably go in the yaml as it's tied to AzDO pools, and in documentation we should specify it as a dependency for osx-arm64.

Reply:
I'm not sure I understand the rationale.  Isn't this file just for installing dependencies for CI.

Reply:
Yeah, this just feels like hiding things in scripts. This is not an installation. At least it's not obvious to me that an installation script hardcodes a path to something that's only true in AzDO (for example, what if a customer runs this expecting to brew install the deps in their box).

Reply:
@hoyosjs & I are in disagreement here...  Any other opinions...  Before we try to drive to consensus?

Reply:
The build step that calls this is called `Install Build Dependencies`.  It seems logical to configure dependencies here too.  If we must I can add another configure dependencies step, but it seems overkill.

Reply:
I'm also OK with renaming the step/script to something like "setup-azdo-agent.sh". My only problem is that: it's now forcing the AzDO layout for Xcode. The script's name should say that.

Reply:
`azdo` seems overly specific.  Are you OK with `setup-native-dependencies-4ci.sh` & `Setup Build Dependencies`?

Reply:
I chose azdo just because it's that: the xcode path for AzDO agents. It's not the same path as helix test agents necessarily or anything (my box doesn't have that path even though I use the same version of XCode now). However, I don't feel too strongly, mostly care about "setup" and "ci/not-local".

Reply:
Wouldn't be people that run this to help them set native dependencies locally?

Reply:
I hadn't seen any instructions suggesting using this for developer setup.

I guess since this is likely temporary I can just make it a separate build step.

Reply:
I added `azDO` to the CI `install-native-dependencies.sh` command line and made the `xcode-select` conditional on the `azDO` argument.  I also added brief header comments.

I initial went down the path of renaming `install` -> `setup`, but I felt like it added confusion.  `setup` in my mind doesn't imply `install`.  However `setup`/`configurations` is a subset of `install`.  So I reverted it all and went with this approach.

Hope this is a satisfactory compromise.